internal/controller/deployment_ to_pdb_controller.go

```diff
@@ -0,0 +1,166 @@
+			Selector:     &metav1.LabelSelector{MatchLabels: deployment.Spec.Selector.MatchLabels},
+		},
+	}
+
+	log.Info("Creating PodDisruptionBudget", "namespace", pdb.Namespace, "name", pdb.Name)
+	if err := r.Create(ctx, pdb); err != nil {
+		log.Error(err, "unable to create PDB")
+		return reconcile.Result{}, err
+	}
+
+	return reconcile.Result{}, nil
+}
+
+func (r *DeploymentToPDBReconciler) generatePDBName(deploymentName string) string {
+	return deploymentName + "-pdb"
+}
+
+// handleDeploymentDeletion deletes the associated PDB when the Deployment is deleted
+func (r *DeploymentToPDBReconciler) handleDeploymentDeletion(ctx context.Context, req ctrl.Request) (reconcile.Result, error) {
+	log := log.FromContext(ctx)
+
+	// Try to get the associated PDB
+	pdb := &policyv1.PodDisruptionBudget{}
+	err := r.Get(ctx, client.ObjectKey{
+		Namespace: req.NamespacedName.Namespace,
+		Name:      r.generatePDBName(req.NamespacedName.Name),
+	}, pdb)
+
+	if err != nil {
+		// If there's no PDB or it can't be fetched, just return
+		log.Info("PodDisruptionBudget does not exist or error fetching", "namespace", req.NamespacedName.Namespace, "name", req.NamespacedName.Name)
+		return reconcile.Result{}, client.IgnoreNotFound(err)
+	}
+
+	// If the PDB exists, delete it
+	log.Info("Deleting PodDisruptionBudget", "namespace", pdb.Namespace, "name", pdb.Name)
+	if err := r.Delete(ctx, pdb); err != nil {
+		log.Error(err, "unable to delete PDB")
+		return reconcile.Result{}, err
+	}
+
+	return reconcile.Result{}, nil
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *DeploymentToPDBReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	logger := mgr.GetLogger()
+	// Set up the controller to watch Deployments and trigger the reconcile function
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&v1.Deployment{}).
+		WithEventFilter(predicate.Funcs{
+			// Only trigger for Create and Delete events
+			CreateFunc: func(e event.CreateEvent) bool {
+				logger.Info("Create event detected, pdb will be created if not exists")
+				// Handle create event (this will be true for all create events)
+				//creationTime, _ := time.Parse(time.RFC3339, e.Object.GetCreationTimestamp().String())
+				//if time.Since(creationTime) < 5*time.Minute {
+				//	return false // Ignore create if it's an existing resource (e.g., by checking timestamp)
+				//}
+				return true // Allow to create for new resources
+			},
+			DeleteFunc: func(e event.DeleteEvent) bool {
+				logger.Info("Delete event detected, pdb will be deleted if exists")
+				// Handle delete event (this will be true for all delete events)
+				return true
+			},
+			UpdateFunc: func(e event.UpdateEvent) bool {
+				logger.Info("Update event detected, no action will be taken")
+				// No need to handle update event
+				//ToDo: distinguish scales from our pdbwatcher from scales from other owners and keep minA	vailable up near replicas.
```

Comment:
You can do this like this (see node reconciler) 

UpdateFunc: func(ue event.UpdateEvent) bool {
				oldNode := ue.ObjectOld.(*v1.Node)
				newNode := ue.ObjectNew.(*v1.Node)
				return oldNode.Spec.Unschedulable == newNode.Spec.Unschedulable
			},
			
			except you need to cast to deployment and check replias instead of node unschedulable